game-app/game-core/src/main/java/games/strategy/triplea/attachments/CanalAttachment.java

```diff
@@ -84,13 +84,17 @@ private void resetCanalName() {
     canalName = "";
   }
 
-  private void setLandTerritories(final String landTerritories) {
+  private void setLandTerritories(final String landTerritories) throws GameParseException {
     final Set<Territory> terrs = new HashSet<>();
-    for (final String name : splitOnColon(landTerritories)) {
-      final Territory territory = getData().getMap().getTerritoryOrNull(name);
-      if (territory == null) {
-        throw new IllegalStateException("Canals: No territory called: " + name + thisErrorMsg());
-      }
+    for (final String territoryName : splitOnColon(landTerritories)) {
+      final Territory territory =
+          getTerritory(territoryName)
+              .orElseThrow(
+                  () ->
+                      new GameParseException(
+                          MessageFormat.format(
+                              "TerritoryAttachment: Setting landTerritories with value {0} not possible; No territory found for {1}",
```

Comment:
Swap the print order for the two variables here (landterritories & territoryName)
That way the "missing territory" is listed first (which will help in case the land territory list is very long and pushes the rest of the message off-screen)